ui/src/settings/segmenters/components/form/validation/schema.js

```diff
@@ -55,13 +55,35 @@ const optionsSchema = yup
     }
   );
 
+const validateArrayString = (arraySchema, arrayName) => yup
+  .string()
+  .test(
+    `${arrayName} valid JSON array`,
+    `${arrayName} must be a valid JSON array`,
+    (array) => {
+      if (array !== "") {
+        try {
+          var parsedArray = JSON.parse(array);
+          if (typeof parsedArray != "array" && !Array.isArray(parsedArray)) {
```

Comment:
sorry for my limited knowledge in javascript, but isn't `typeof (array)` is `object`?

Reply:
Yes you're right `typeof someArray` always returns an object. I modified this check incorrectly 😅. I've changed it to `typeof parsedArray != "object" || !Array.isArray(parsedArray)` now, which is similar to what we already have for some of the other tests in this file.